src/Bonus.java

```diff
@@ -1,66 +1,86 @@
-        - Maak als eerst een String variabele met als waarde een lege String. (of gebruik een StringBuilder)
-        - Schrijf vervolgens een for-loop om de items in de hashset een voor een aan de String variabele toe te voegen.
-        - Return de (gevulde) String variabele
-         */
-    }
+    public static void feedback(String stringNumber) {
+        Scanner scanner = new Scanner(System.in);
+        boolean play = true;
+        String invalidInput = "Ongeldige invoer";
 
+        while (play) {
+            System.out.println("\nType 's' om te stoppen \nType 'g' om te gokken");
 
+            String input = scanner.nextLine();
 
-    public static void/*moet dit "void" zijn of wat anders?*/ feedback(String stringnumber) {
-        Scanner scanner = new Scanner(System.in);
-        StringBuilder feedback = new StringBuilder();
-        System.out.println("+ = juiste nummer op de juiste plek, O = juiste nummer verkeerde plek, X = verkeerde nummer");
-        System.out.println("Doe een gok, Let op vul 4 getallen in.");
-        String guess = scanner.nextLine();
-        if (Objects.equals(guess, stringnumber)) {
-            System.out.println("gefeliciteerd je hebt het goed");
-        } else {
-            for (int i = 0; i < 4; i++) {
-                if (guess.substring(i, i + 1).equals(stringnumber.substring(i, i + 1))) {
-                    feedback.append("+");
-                } else if (stringnumber.contains(guess.substring(i, i + 1))) {
-                    feedback.append("0");
+            if (input.equals("s")) {
+                play = false;
+                System.out.println("Spel gestopt.");
+            } else if (input.equals("g")) {
+                System.out.println("+ = juiste nummer op de juiste plek, O = juiste nummer verkeerde plek, X = verkeerde nummer");
+                System.out.println("Doe een gok, Let op: vul 4 getallen in.");
+                String guess = scanner.nextLine();
+
+                if (guess.length() != 4) {
+                    System.out.println("Ongeldige invoer, vul precies 4 cijfers in.");
+                    continue;
+                }
+
+                if (Objects.equals(guess, stringNumber)) {
+                    System.out.println("Gefeliciteerd! Je hebt het goed geraden: " + stringNumber);
+                    play = false;
                 } else {
-                    feedback.append("X");
+                    StringBuilder feedback = new StringBuilder();
+
+                    for (int i = 0; i < 4; i++) {
+                        if (guess.charAt(i) == stringNumber.charAt(i)) {
+                            feedback.append("+");
+                        } else if (stringNumber.contains(String.valueOf(guess.charAt(i)))) {
+                            feedback.append("O");
+                        } else {
+                            feedback.append("X");
+                        }
+                    }
+
+                    System.out.println(feedback.toString());
```

Comment:
You can remove .toString() here since, it's redundant. Here's why:

1. **`System.out.println(Object)`**: When you pass an object (in this case, a `StringBuilder`) to `System.out.println()`, the `println()` method will automatically call the `toString()` method on that object. For `StringBuilder`, this is already overridden to return the accumulated string.

2. **Redundant call**: So when you write `System.out.println(feedback.toString());`, it's explicitly calling `toString()` on `feedback`, but you could just pass `feedback` directly and let `println()` handle it, making the `.toString()` call unnecessary.

You can simplify the line like this:
```java
System.out.println(feedback);  // `println()` automatically calls `toString()` here
```

Reply:
Bedankt voor de uitleg! Ik kreeg deze melding al in IntelliJ, maar begreep het niet. Dus bedankt om het uit te leggen. 😄